src/hammer-vlsi/hammer_vlsi/cli_driver.py

```diff
@@ -856,6 +859,77 @@ def auto_action(driver: HammerDriver, append_error_func: Callable[[str], None])
 
         return driver, errors
 
+    @staticmethod
+    def make_build_inputs(driver: HammerDriver, append_error_func: Callable[[str], None]) -> Optional[dict]:
+        """
+        TODO
+        """
+        build_system_key = "vlsi.core.build_system"
+        build_system = str(driver.database.get_setting(build_system_key, "none"))
+        dependency_graph = driver.get_hierarchical_dependency_graph()
+        if build_system == "none":
+            pass
+        elif build_system == "make":
```

Comment:
We might want to make these functions then index into it like `build_systems[system]()`

Reply:
IMO we should make that change whenever another build system is supported, which I imagine will be a while unless you guys are getting bazel set up.

Reply:
Are there strong interdependencies between the tools? My thought was that it would make future PRs to add other `build_system`s less invasive.

Reply:
Maybe I'm misunderstanding what you're proposing. I was suggesting something like

```
elif build_system == "bazel":
    # Do stuff here
```

That's more or less the pattern we've used for other mode-type things before. I am interpreting what you want as:

```
build_systems[build_system](dependency_graph)
```

which doesn't seem any more or less invasive, but slightly less easy to read. If you want something different can you write a quick snippet?

Reply:
We might imagine something like

```python
def makefile(...): ...

def bazel(...): ...

def scons(...): ...

build_systems = {
  "makefile": makefile,
  "bazel": bazel,
  "scons": scons,
  ...
} # <-- this may be even programmatically modified
```

Reply:
So the value proposition is that you can dynamically edit `build_systems` (and not paint yourself into a corner like we did with `submit_command`)?

Reply:
It seems so. I think it also helps since adding a new system now just involves

```python
def my_new_system(...): ...

build_systems = { 
  # ...
  "my_new_system": my_new_system
}
```

I think for modes it makes sense to just do `if/elif` since it's not often that we add new modes

Reply:
Lemme give it a shot

Reply:
I think this is resolved, but let me know if you have other feedback